octodns/zone.py

```diff
@@ -140,6 +140,11 @@ def changes(self, desired, target):
                                record)
                 changes.append(Delete(record))
             else:
+                if not target.supports(record):
+                    self.log.debug('changes:  skipping record=%s %s - %s does '
+                                   'not support it', record.fqdn, record._type,
+                                   target.id)
+                    continue
```

Comment:
This prevents changes that are unsupported in the target provider from being calculated. Since the `supports` method on `BaseProvider` now checks that root NS record management is supported and enabled, this will filter out changes to root NS records unless actually desired and supported. This also makes all the existing tests pass since none of the behavior changes. I've added some new tests to check the `manage_root_ns` functionality as well.